x-pack/plugins/fleet/public/applications/fleet/hooks/use_ui_extension.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext } from 'react';
+import { UIExtensionPoint, UIExtensionsStorage } from '../../../../common/types/ui_extensions';
+
+export const UIExtensionsContext = React.createContext<UIExtensionsStorage>({});
+
+type SpecificExtensionPoint<A, T, V> = A extends { type: T; view: V } ? A : never;
+type NarrowViews<T = UIExtensionPoint['type'], A = UIExtensionPoint> = A extends {
+  type: T;
+}
+  ? A
+  : never;
+
+export const useUIExtension = <
+  T extends UIExtensionPoint['type'] = UIExtensionPoint['type'],
+  V extends NarrowViews<T>['view'] = never
+>(
+  integration: UIExtensionPoint['integration'],
+  type: T,
+  view: V
+): SpecificExtensionPoint<UIExtensionPoint, T, V>['component'] | undefined => {
+  const extension = useContext(UIExtensionsContext)?.[integration]?.[type]?.[view];
```

Comment:
I have a personal preference to throw when a required context is not present. I found it make debugging easier

Reply:
Good point on it being easier to debug. Will make the change.